### DISCUSSION TITLE
Support await and done syntax at the same time

### DISCUSSION THREAD

**Discussion by molszanski** [vitest-dev/vitest#1699] [Q&A]
Assume I have code like this:
```ts
it("should be able subscribe to container set change", async (cb) => {
  const cont = getMainMockAppContainer()
  let containerSet = await cont.getContainerSet(["aCont", "bCont", "cCont"])

  expect(containerSet.bCont.b2).toMatchObject({ a1: {} })
  expect(containerSet.cCont.c2.size).toBe(5)

  cont.subscribeToContinerSet(["aCont", "bCont", "cCont"], (containerSet) => {
    expect(containerSet.cCont.c2.size).toBe(10)
    cb()
  })

  containerSet.cCont.upgradeCContainer()
})
```

Basically I am using an `await` and `done()` callback style at the same time. 

Is there a way to make it work with vitest?

**Comment by sheremet-va** [comment]
https://vitest.dev/guide/migration.html#migrating-from-jest

```diff
- it('should work', (done) => {
+ it('should work', () => new Promise(done => {
    // ...
    done()
- })
+ }))
```

**Reply by molszanski** [reply]
Yeah, Ii've tried it, no luck
```ts
it("This works", () =>
  new Promise((done) => {
    let a = 12
    expect(a).toBe(12)
    // @ts-ignore
    done()
  }))

it("This doesn't", () =>
  new Promise(async (done) => {
    let a = 12
    expect(a).toBe(13)
    // @ts-ignore
    done()
  }))
```
Making stackblitz playground

**Reply by molszanski** [reply]
https://stackblitz.com/edit/node-rjjsul?file=lol-test.test.js

**Reply by molszanski** [reply]
```ts

it("This doesn't", () =>
  new Promise((done) => {
    setTimeout(() => {
      let a = 12
      expect(a).toBe(13)
      done()
    }, 100)
  }))
```

**Reply by sheremet-va** [reply]
Well, just call async functions outside of new Promise.

**Reply by molszanski** [reply]
@sheremet-va, hmm, I don't understand :( 

There is a not-so-rare pattern in tests suites where you wait for some event and only then check results and end the test. 

Not sure how to make it happen

**Comment by molszanski** [comment]
Any ideas? :/ This blocks me from moving some projects to vitest. 

I am happy to create a PR, but trying to figure out if this is something worth doing within the existing architecture
